### PR TITLE
Change admin password, routes protected, logout button fixed

### LINKED ISSUE #123: Frontend: Change password page for admin, logout button and permissions
### Describe the feature

Making possible for the admin to change his password.
Some pages should ot be accessible by normal users.

### Use Case

As an admin i want to change my credentials to not use the default ones.

### Proposed Solution

When admin logs in for the first time with default credentials, he should be prompted to change his password.

### Other Information

_No response_

### PR DESCRIPTION
**Pull request by MiguelCruzeiro**
### Issue

Closes #123 .

### Reason for this change

Admin should not use default credentials, logout button not working as expected, routes not protected.

### Description of changes

Routes protected depending on user role, logout button not appearing when not logged in, admin now prompted to change his password when loggin in with default credentials.

### Description of how you validated changes

Testing manually.

### Checklist
- [x] I have tested my changes locally.
- [ ] I have updated the documentation to reflect my changes.
- [ ] I have added/updated unit tests (if applicable).
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://ua-smart-signage-platform.github.io/Documentation-Website/guidelines/)

### CODE REVIEW

File: frontend/src/components/NavBar/index.jsx [hunk 1]
```diff
@@ -14,12 +14,19 @@ function NavBar() {
...
     const changeTheme = useThemeStore((state)=> state.changeTheme)
     const navigate = useNavigate();
+    const isLoggedIn = localStorage.getItem('access_token');
+    const userInfo = localStorage.getItem('userInfo');
+    const userRole = userInfo ? JSON.parse(userInfo).role : null;
+
 
     const handleLogout = () => {
+        console.log("user role ", userRole)
+        console.log("user info ", userInfo)
```
Comment (pdPinho) [inline frontend/src/components/NavBar/index.jsx]: Don't forget to remove this

File: frontend/src/components/NavBar/index.jsx [hunk 2]
```diff
@@ -14,12 +14,19 @@ function NavBar() {
...
         localStorage.removeItem('access_token');
         localStorage.removeItem('id_token');
         localStorage.removeItem('refresh_token');
         localStorage.removeItem('userInfo');
+
```
Comment (pdPinho) [inline frontend/src/components/NavBar/index.jsx]: New line that might not be needed

File: frontend/src/index.jsx
Comment (pdPinho) [inline frontend/src/index.jsx:1]: What happened on this page? Why is its format different?

File: frontend/src/pages/Wso2Login/index.jsx [hunk 3]
```diff
@@ -40,10 +42,19 @@ const Wso2Login = () => {
       const user_data = { username: response.data.username, role: response.data.role };
       
       setWithExpiry('access_token', jwt, 3600 * 1000 * 10);
-      localStorage.setItem('userInfo', user_data);
+      localStorage.setItem('userInfo', JSON.stringify(user_data));
       console.log('userInfo', user_data);
+      console.log('user_role', user_data.role);
```
Comment (pdPinho) [inline frontend/src/pages/Wso2Login/index.jsx]: remove this

File: frontend/src/pages/Wso2Login/index.jsx [hunk 4]
```diff
@@ -40,10 +42,19 @@ const Wso2Login = () => {
...
         
-      navigate("/dashboard");
+      if (password === DEFAULT_PASSWORD) {
+        // Redirect to change password page if it's the first login
+        navigate("/change-password");
+        setShowWarning(true);
+        return
+      } else {
+        console.log('userRole', localStorage.getItem('userInfo').role);
```
Comment (pdPinho) [inline frontend/src/pages/Wso2Login/index.jsx]: remove this

File: frontend/src/index.jsx
Comment (fungame2270) [inline frontend/src/index.jsx:1]: Esta pagina foi alterada neste pull requeste para a versao mais antiga do reactRouter. Se fores a documentaçao do react router vais notar que eles no tutorial mostram o novo metodo de fazer routes usando o CreateBrowserRouter que era o q estava a ser usado.

**Review by fungame2270 — APPROVED** [review]
thanks

**Pull request merged**